segfast/memmap_loader.py

```diff
@@ -214,7 +214,13 @@ def callback(future, start):
                     future = executor.submit(read_chunk, path=self.path, shape=self.n_traces,
                                              offset=self.file_traces_offset, mmap_dtype=mmap_trace_dtype,
                                              buffer_dtype=dst_headers_dtype, headers=headers, indices=chunk_indices)
-                    future.add_done_callback(partial(callback, start=i * chunk_size))
+
+                    future.add_done_callback(partial(callback, start=start))
+
+                    if isinstance(chunk_indices, slice):
+                        start += chunk_size
+                    else:
+                        start += len(chunk_indices)
```

Comment:
Probably, can adapt the code from `seismiqb` / `SeismicPro`:
```python
n_chunks, last_chunk_size = divmod(n, chunk_size)
chunk_sizes = [chunk_size] * n_chunks
if last_chunk_size:
    chunk_sizes += [last_chunk_size]
    n_chunks += 1

chunk_starts = np.cumsum([0] + chunk_sizes[:-1])
chunk_ends = np.cumsum(chunk_sizes)
```

This way, both `starts` and `ends` are defined and can be iterated over

Reply:
`chunk_starts/chunk_end` makes better in the case when you work with "continuous" or, at least, "regular" arrays. In that situation, I guess, it's a nice idea to leave `slices` for `indices=None` if it's faster than dealing with `indices=np.arange(self.n_traces)`.

Reply:
Probably the next option is nicer than the previous one.